internal/oci/runtime_oci.go

```diff
@@ -956,11 +956,16 @@ killContainer:
 	// We cannot use ExponentialBackoff() here as its stop conditions are not flexible enough.
 	kwait.BackoffUntil(func() {
 		if _, err := r.runtimeCmd("kill", c.ID(), "KILL"); err != nil {
+			if errors.Is(err, unix.ESRCH) {
+				// Process is already gone, stop the loop.
+				stop()
+				return
+			}
 			log.Errorf(ctx, "Killing container %v failed: %v", c.ID(), err)
 		}
-
 		if err := c.Living(); err != nil {
```

Comment:
Are there issues with `Living()` not working currently? The idea is for it to ascertain whether a process is still "alive" and break the kill loop if not.

Reply:
The issue appears to be related to the timing and sequence of operations. Specifically, `r.runtimeCmd("kill", c.ID(), "KILL")` successfully kills the container, but CRI-O waits for a response from crun. During this time, crun may have already deleted the container, leading to the "No such process" error. The error occurs because the container process is already gone when CRI-O checks its status.